src/actor/calc/lookup.rs

```diff
@@ -0,0 +1,138 @@
+use super::super::stat::Stat;
+
+#[allow(dead_code)]
+pub enum LevelColumn {
+    MP,
+    MAIN,
+    SUB,
+    DIV,
+    HP,
+    ELMT,
+    THREAT,
+}
+
+// Assume level 80
+// https://www.akhmorning.com/allagan-studies/modifiers/
+#[allow(dead_code)]
+pub fn level_modifiers(column: LevelColumn) -> i64 {
+    match column {
+        LevelColumn::MP => 10000,
+        LevelColumn::MAIN => 340,
+        LevelColumn::SUB => 380,
+        LevelColumn::DIV => 3300,
+        LevelColumn::HP => 4400,
+        LevelColumn::ELMT => 0, // ??? on akhmorning. https://www.akhmorning.com/allagan-studies/modifiers/levelmods/
+        LevelColumn::THREAT => 569,
+    }
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Copy, Clone)]
+pub enum Job {
+    GLA,
+    PGL,
+    MRD,
+    LNC,
+    ARC,
+    CNJ,
+    THM,
+    PLD,
+    MNK,
+    WAR,
+    DRG,
+    BRD,
+    WHM,
+    BLM,
+    ACN,
+    SMN,
+    SCH,
+    ROG,
+    NIN,
+    MCH,
+    DRK,
+    AST,
+    SAM,
+    RDM,
+    BLU,
+    GNB,
+    DNC,
+    None,
+}
+
+impl Job {
+    pub fn primary_stat(&self) -> Stat {
+        match self {
+            Job::LNC | Job::PGL | Job::DRG | Job::MNK | Job::SAM => Stat::Strength,
+            Job::ARC | Job::ROG | Job::BRD | Job::NIN | Job::MCH | Job::DNC => Stat::Dexterity,
+            Job::GLA | Job::MRD | Job::PLD | Job::WAR | Job::DRK | Job::GNB => Stat::Strength,
+            Job::THM | Job::ACN | Job::BLM | Job::SMN | Job::RDM | Job::BLU => Stat::Intelligence,
+            Job::CNJ | Job::WHM | Job::SCH | Job::AST => Stat::Mind,
+            _ => panic!("Tried to get primary stat of unknown job: {:?}", self),
+        }
+    }
+
+    pub fn trait_multiplier(&self) -> i64 {
+        match self {
+            Job::ARC | Job::BRD | Job::MCH | Job::DNC => 120,

```

Comment:
In general, large look-up tables may be more reasonable to store as data and then load. The current approach is fine however.